editor/editor_translation_parser.cpp

```diff
@@ -37,15 +37,30 @@
 
 EditorTranslationParser *EditorTranslationParser::singleton = nullptr;
 
-Error EditorTranslationParserPlugin::parse_file(const String &p_path, Vector<String> *r_extracted_strings) {
+Error EditorTranslationParserPlugin::parse_file(const String &p_path, Vector<String> *r_ids, Vector<Vector<String>> *r_ids_ctx_plural) {
 	if (!get_script_instance())
 		return ERR_UNAVAILABLE;
 
 	if (get_script_instance()->has_method("parse_file")) {
-		Array extracted_strings;
-		get_script_instance()->call("parse_file", p_path, extracted_strings);
-		for (int i = 0; i < extracted_strings.size(); i++) {
-			r_extracted_strings->append(extracted_strings[i]);
+		Array ids;
+		Array ids_ctx_plural;
+		get_script_instance()->call("parse_file", p_path, ids, ids_ctx_plural);
+
+		// Add user's extracted translatable messages.
+		for (int i = 0; i < ids.size(); i++) {
+			r_ids->append(ids[i]);
+		}
+
+		// Add user's collected translatable messages with context or plurals.
+		for (int i = 0; i < ids_ctx_plural.size(); i++) {
+			Array arr = ids_ctx_plural[i];
+			ERR_FAIL_COND_V_MSG(arr.size() != 3, ERR_INVALID_DATA, "Array entries written into msgids_ctx_plural of EditorTranslationParserPlugin parse_file() method should have the form [\"message\",\"context\",\"plural message\"]");
```

Comment:
```suggestion
			ERR_FAIL_COND_V_MSG(arr.size() != 3, ERR_INVALID_DATA, "Array entries written into `msgids_context_plural` in `parse_file()` method should have the form [\"message\", \"context\", \"plural message\"]");
```
Removed the class name to make it shorter, and it should already be clear from the location of the error message I think.

Reply:
You should add clearer documentation about this requirement in the XML doc itself, it's not something users should have to discover through an error (even though it's partly covered in the examples, but it's not explained that it's a hard requirement and what each argument means).